590and592

jdlrobson opened the pull request:
Apologies for the size of this branch.
What started off as addressing 590 and 592 found similar problems in the change password, members forms as well as spaces and inclusions.

The problem was not simple to address, and it was in fact easier to refactor all these plugins in a similar way to which has been done for the login and registration forms.

All tests remain passing, the code is hopefully a lot cleaner to read now and has less bloat.

jdlrobson committed:
added displayError to TiddlySpaceForms

provides what will hopefully be an easier way to do error
handling.

jdlrobson committed:
added checkbox support

jdlrobson committed:
refactored to make use of TiddlySpaceForms for form creation

this takes care of a lot of the ui this plugin introduced,
via some renaming and movement of templating to js

jdlrobson committed:
refactored to make error message work with formMaker extension

Note this removes the error message, but currently doesn't work
as the form is not made via the formMaker extensions.

jdlrobson committed:
removed references to active parameter

There seems to be dead code here which is not exposed anywhere in the
ui. It provided a drop down of your spaces which could be selected for
including.  Thus removing.

jdlrobson committed:
create form with formMaker

this makes the error handling work, and also cleans up the code
somewhat. The template is now js rather than part of the plugin
slices.

jdlrobson committed:
tweak lingo

Updated the in progress message and submit label
Removed a no longer needed message for active

jdlrobson committed:
refactored form submit handler

jdlrobson committed:
replaced TiddlySpaceLogin displayError with formMaker displayError

This removes the bad reference javascript error.
Will do not do anything with out refactoring form creation.

jdlrobson committed:
refactored form creation

the add members form is now created via config.extensions.formMaker

jdlrobson committed:
refactored change password error handling

in exactly the same way as in previous commits

jdlrobson committed:
update password helper to allow providing different name

this is useful for the change password form. The inputs are all
passwords but have specific names.

jdlrobson committed:
refactor change password to use formMaker

just as in previous commits

jdlrobson committed:
update requires header

make sure the TiddlySpaceAdmin plugin has been loaded first.

jdlrobson committed:
change test loading order

TiddlySpaceSpaces now depends on TiddlySpaceAdmin

jdlrobson committed:
Merge branch 'master' of git://github.com/TiddlySpace/tiddlyspace into 590and592

jdlrobson committed:
update change password button label

cdent commented:
from irc:
cdent: the 590and592 stuff looks like a good bunch of changes
[5:06pm] cdent: jdlrobson: minor complaint:
[5:06pm] cdent: on the password form the include the current space checkbox is not next to the text
[5:06pm] cdent: it is in the line above
[5:08pm] cdent: also, when submitting the add identity form, the text of the input box is changed to the paramefier

jdlrobson commented:
Yeh I must confess I was aware of the checkbox jumping up a line. I couldn't find a nice way to style it.
Can look into it again on Monday when I'm a little fresh.

Haven't touched the identity stuff, will raise that as an issue however.

jdlrobson committed:
wrap inputs within a div, add label parameter to checkbox

checkboxes could do with being adjacent to their labels rather than
separate. This label parameter provides this.

wrapping elements in a div allows styling such as putting inputs on
new lines or the same line.

jdlrobson committed:
alter form to put label alongside checkbox
